rasa/core/brokers/pika.py

```diff
@@ -193,6 +193,7 @@ async def _connect(self) -> aio_pika.RobustConnection:
                 )
                 await asyncio.sleep(self._retry_delay_in_seconds)
 
+        last_exception = cast(Exception, last_exception)
```

Comment:
Should the `cast` happen in the `except` block as that's when we know that it is no longer `none`?

Reply:
🤔 I don't think I've tried that, let me see if `mypy` picks it up properly